Add instructions for the release process (#346)
kubernetes-sigs/aws-iam-authenticator: docs/RELEASE.md

# AWS IAM Authenticator Release Process

> NOTE: your github user must have the required permissions and you must have generated a github token for that user.

Reviewer: ```suggestion
NOTE: Your GitHub account must have the required permissions and you must have generated a GitHub token.
```

## Choosing the release version and branch

Using semantic versioning, pick a release number that makes sense by bumping the major, minor or patch release version.  If its a major or minor release (backwards incompatible changes, and new features, respectively) then you will want to start this process with an alpha release first.  Here are some examples:

Bumping a minor version after releasing a new feature:
```

> v0.4.5 -> v0.5.0-alpha.0

Reviewer: Can we make these examples 1.x.y? Technically v0.x.y are unstable and can have breaking changes at any time.

Separately, its probably worth cutting a v1.0.0 since we're pretty stable

Author: That's fair on both points.  We can probably work towards a 1.0 but should have a bit of a triage and bug bash before cutting it.

```
./hack/changelog.py --token=$GITHUB_TOKEN --section-title="Release v0.5.2" --range=v0.5.1..90653708db3f6437a446bbeec15b5036db66a855 --print-only
```

After checking for correctness, pass the `--changelog-file` argument to add the new text.
```
./hack/changelog.py --token=$GITHUB_TOKEN --section-title="Release v0.5.2" --range=v0.5.1..90653708db3f6437a446bbeec15b5036db66a855 --changelog-file=CHANGELOG.md
```

Also, bump the image version in `deploy/example.yaml` to the new version.

Push the changes to a branch on your fork, and create a PR against the kubernetes-sigs upstream repository.

## Tagging the release

One the PR merges, pull the master branch locally and tag the release commit with the relase tag.
```
git pull upstream master
git tag v0.5.2
```

## Run goreleaser

In order to run goreleaser to make the release, you'll need to authenticate to the release ECR registry, and then run goreleaser.

```
aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 602401143452.dkr.ecr.us-west-2.amazonaws.com
goreleaser release --rm-dist
```

TODO: configure goreleaser to use `./hack/changelog.py` to format the release text.

> ## Check the release on github

Reviewer: ```suggestion
## Check the release on GitHub
```

> Look at the release that was just published and validate that the release has the appropriate binaries (compare to a previous release).  Check the ECR registry to make sure that the images were published.  Finally, edit the release text to match previous releases, by copying the changelog text and adding the container image links.

Reviewer: Can we document what the proper release binaries are?
